server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/EventDeadLettersIntegrationTest.java

```diff
@@ -309,13 +321,13 @@ public void failedEventShouldBeCorrectlyProcessedByListenerAfterSuccessfulRedeli
             .body("additionalInformation.group", is(SERIALIZED_GROUP))
             .body("additionalInformation.eventId", is(failedEventId));
 
-        assertThat(retryEventsListener.getSuccessfulEvents()).containsOnly(failedEvent);
+        assertThat(retryEventsListener.getSuccessfulEvents()).hasSize(1);
```

Comment:
Task completion IMO does not necessarily mean the listener is fully processed.

That migh be true for the InVnEventBus you are using here but it can't assumed to be true.

Let's rely on `CalmlyAwait` and poll with a timeout until it becomes true.